eks/locals.tf

```diff
@@ -1,5 +1,5 @@
 locals {
-  helm_values = (var.metrics_storage.iam_role_arn != "") ? [{
+  helm_values = var.metrics_storage != null ? ( var.metrics_storage.iam_role_arn != "" ? [{
```

Comment:
I would be more inclined to replace the condition altogether since the IAM role would be required from the moment someone wants to pass a metrics bucket.

```
  helm_values = (var.metrics_storage.iam_role_arn != "") ? [{
    kube-prometheus-stack = {
      prometheus = {
        serviceAccount = {
          annotations = {
            "eks.amazonaws.com/role-arn" = var.metrics_storage.iam_role_arn
          }
        }
      }
    }
  }] : []
```